fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -379,6 +379,9 @@ class RocksDBEventListener : public rocksdb::EventListener {
 	void OnFlushCompleted(rocksdb::DB* db, const rocksdb::FlushJobInfo& info) override { *lastFlushTime = now(); }
 
 private:
+	// lastFlushTime is used by two threads. One Thread is reading the value and the other thread is updating the value.
+	// If the reader thread gets a wrong value due to race, that will be still fine in this case(probably an extra flush
+	// or no flush). Considering the cost of atomic, avoided it here in this case.
```

Comment:
Generally an intentional race condition is tricky to reason about, especially because tools like TSAN would still flag them, so we'd have to find a way to suppress them. We don't use TSAN today but it could be useful in the future especially as rocksdb does background work off the main thread, and we're trying out grpc, etc. 

Curious about the performance cost of atomic here, a good first order approximation is around ~10ns (upper bound could be around 100ns, depending on various factors e.g. if compiler uses atomic CAS and that has contention, there will be retries). So unless we're writing/reading `lastFlushTime` a lot of times and that too in the critical path, it should be ok to use. As an example, if you have a tight loop in the critical path which uses atomic, and that loop has 1M iterations/sec, the wallclock time overhead could be up 10-100ms, which is very high. But my understanding is that here `lastFlushTime` would be written by rocksdb event listener callback (amortized over 1 sec, the number of times we flush should be way less than 1), and `lastFlushTime` would be read roughly every `ROCKSDB_MANUAL_FLUSH_TIME_INTERVAL (10 seconds?)`, so overall we're looking at a few ns overhead amortized over a second. 

Thought to share in case it helps evaluating the tradeoffs.